Text/Pandoc/Definition.hs

```diff
@@ -321,24 +324,50 @@ instance ToJSON Meta
 
 instance FromJSON CitationMode
   where parseJSON = parseJSON'
-instance ToJSON CitationMode
-  where toJSON = toJSON'
+instance ToJSON CitationMode where
+  toJSON cmode =
+    object [ "t" .= String s
```

Comment:
I think it would be a good idea to abstract out this pattern, to avoid all the code repetition.   Also, is there a reason not to use Show instances?  Then this could be something like

    toJSON cmode = tagged (show cmode) Nothing

In cases where we have a nonempty content, we could have (e.g.)

    toJSON (Emph ils) = tagged "Emph" ils

This would make the code more compact and decrease the possibility of small random errors.